core/templates/domain/admin/admin-backend-api.service.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Backend api service for fetching the admin data;
+ */
+
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+import {
+  IAdminDataBackendDict,
+  AdminData,
+  AdminDataObjectFactory
+} from 'domain/admin/admin-data-object.factory';
+import { AdminPageConstants } from
+  'pages/admin-page/admin-page.constants';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AdminBackendApiService {
+  constructor(
+    private http: HttpClient,
+    private adminDataObjectFactory: AdminDataObjectFactory) {}
+
+  getData(): Promise<AdminData> {
```

Comment:
This returns all the data needed by the admin page. imo `AdminData` seems fine!

Reply:
`AdminData` name seems weird to me Like AdminData isn't related to admin and we expect a domain class to store data then a `Data` prefix is not required. I'm not sure about a better name for this.

@seanlip Do you have any suggestion for a class name which stores these data:
```
'config_properties': ...
'continuous_computations_data': ...
'demo_collections': ...
'demo_explorations': ...
'demo_exploration_ids': ...
'human_readable_current_time': ...
'one_off_job_specs': ...
'audit_job_specs': ...
'recent_job_data': ...
'unfinished_job_data': ...
'updatable_roles': ...
'viewable_roles': ...
'topic_summaries': ...
'role_graph_data': ...
```

Note: These values are used for the loading admin page.

Reply:
Hm, I think the issue with this is that it's really not all just one "thing". In general, it should be fine for a GET request to return multiple "things". I.e., it shouldn't be the case that, just because a GET request returns a bunch of things, that they are all the same thing conceptually -- we sometimes do just one request for multiple things, in order to save bandwidth.

@nishantwrp is it possible for this concept to be reflected in the types? I.e., to convey that getData() returns a dict with various key-value pairs, where different keys correspond to different items (some of which may be strings, some of which may be domain objects, etc.). I don't think "AdminData" really makes sense as a long-standing concept.

/cc @vojtechjelinek

Reply:
From the discussion:

- It shouldn't be the case that, just because a GET request returns a bunch of things, that they are all the same thing conceptually. We sometimes do just one request to save bandwidth.
- if return multiple different chunks of data then make the succesCallback look like this: successCallback(chunk1, chunk2 ...)?

Reply:
Hmm, then I think we can remove the `AdminData` object factory because `AdminData` shouldn't be a `class`. So, we can just camelize the `keys` of the backend dict and return it. And the name of that interface could be `AdminPageData`?

/cc @seanlip

Reply:
I'm not totally sure what you're proposing exactly, I'd have to see how it "looks" before making a comment. Maybe make the changes to the PR itself and then @-mention us in those to take a look?

I agree that AdminData doesn't make sense conceptually as a class but I don't really have a strong sense yet of what the actual conventions should be (sorry).

Reply:
Made the changes. ptal!

Reply:
Hi @nishantwrp, I think this looks fine to me -- or, at least, I can't really think of a better alternative. @DubeySandeep @vojtechjelinek what do you think?

Reply:
LGTM!